### ISSUE TITLE
longhorn-driver-deployer v0.7.0 CrashLoopBackOff on fresh install

### ISSUE THREAD

**Issue by cameronbraid** [longhorn/longhorn#1072]
installed via
```
kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/longhorn.yaml
```

host `Ubuntu 18.04.3 LTS`

root@node01 ~# kubectl version
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:07:13Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```

root@node01 ~# docker images|grep long
```
longhornio/longhorn-manager                                    v0.7.0              ad41346e2bd1        3 months ago        261MB
longhornio/longhorn-engine                                     v0.7.0              a135d79e02ea        3 months ago        307MB
```

root@node01 ~# kubectl get pods --namespace longhorn-system
```
NAME                                        READY   STATUS             RESTARTS   AGE
engine-image-ei-ec95b5ad-k6nkc              1/1     Running            0          30m
engine-image-ei-ec95b5ad-mgc2x              1/1     Running            0          30m
engine-image-ei-ec95b5ad-p5prl              1/1     Running            0          30m
instance-manager-e-b246f54d                 1/1     Running            0          27m
instance-manager-e-f0b16ad4                 1/1     Running            0          27m
instance-manager-e-f1d23112                 1/1     Running            0          27m
instance-manager-r-4dfc47b1                 1/1     Running            0          27m
instance-manager-r-563e060e                 1/1     Running            0          27m
instance-manager-r-592b84d5                 1/1     Running            0          27m
longhorn-driver-deployer-5558df9859-s9n7s   0/1     CrashLoopBackOff   8          34m
longhorn-manager-ghhzf                      1/1     Running            1          34m
longhorn-manager-jjpbf                      1/1     Running            4          34m
longhorn-manager-zl4zw                      1/1     Running            2          34m
longhorn-ui-b97b74b8-8hjn6                  1/1     Running            0          34m
```

root@node01 ~# kubectl -n longhorn-system logs longhorn-driver-deployer-5558df9859-s9n7s
```
time="2020-02-26T15:01:22Z" level=debug msg="Deploying CSI driver"
time="2020-02-26T15:01:22Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:24Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:25Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:27Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:28Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:29Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:30Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:31Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:32Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:33Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:34Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:35Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:36Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:37Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:38Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:40Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-02-26T15:01:41Z" level=info msg="Proc found: kubelet"
time="2020-02-26T15:01:42Z" level=info msg="Try to find arg [--root-dir] in cmdline: [/usr/bin/kubelet --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --config=/var/lib/kubelet/config.yaml --cgroup-driver=systemd --network-plugin=cni --pod-infra-container-image=k8s.gcr.io/pause:3.1 --resolv-conf=/run/systemd/resolve/resolv.conf --node-ip=10.111.0.183 --max-pods=500 ]"
time="2020-02-26T15:01:42Z" level=warning msg="Cmdline of proc kubelet found: \"/usr/bin/kubelet\x00--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf\x00--kubeconfig=/etc/kubernetes/kubelet.conf\x00--config=/var/lib/kubelet/config.yaml\x00--cgroup-driver=systemd\x00--network-plugin=cni\x00--pod-infra-container-image=k8s.gcr.io/pause:3.1\x00--resolv-conf=/run/systemd/resolve/resolv.conf\x00--node-ip=10.111.0.183\x00--max-pods=500\x00\". But arg \"--root-dir\" not found. Hence default value will be used: \"/var/lib/kubelet\""
time="2020-02-26T15:01:42Z" level=info msg="Detected root dir path: /var/lib/kubelet"
time="2020-02-26T15:01:42Z" level=info msg="Upgrading Longhorn related components for CSI v1.1.0"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1480fb2]

goroutine 1 [running]:
github.com/longhorn/longhorn-manager/app.upgradeLonghornRelatedComponents(0xc0000e0000, 0xc00004200e, 0xf, 0x0, 0x0)
        /go/src/github.com/longhorn/longhorn-manager/app/upgrader.go:33 +0x252
github.com/longhorn/longhorn-manager/app.deployCSIDriver(0xc0000e0000, 0xc000477820, 0xc0004acc60, 0x7ffeafd21b29, 0x22, 0x7ffeafd21b5a, 0x1f, 0x184cc13, 0x9)
        /go/src/github.com/longhorn/longhorn-manager/app/driver.go:193 +0x575
github.com/longhorn/longhorn-manager/app.deployDriver(0xc0004acc60, 0x5, 0x0)
        /go/src/github.com/longhorn/longhorn-manager/app/driver.go:142 +0x38b
github.com/longhorn/longhorn-manager/app.DeployDriverCmd.func1(0xc0004acc60)
        /go/src/github.com/longhorn/longhorn-manager/app/driver.go:98 +0x2f
github.com/urfave/cli.HandleAction(0x158b0c0, 0x18f3b58, 0xc0004acc60, 0xc0004acc60, 0x0)
        /go/src/github.com/longhorn/longhorn-manager/vendor/github.com/urfave/cli/app.go:525 +0x127
github.com/urfave/cli.Command.Run(0x1851197, 0xd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/longhorn/longhorn-manager/vendor/github.com/urfave/cli/command.go:174 +0x51c
github.com/urfave/cli.(*App).Run(0xc00003c1c0, 0xc0000b8000, 0x7, 0x7, 0x0, 0x0)
        /go/src/github.com/longhorn/longhorn-manager/vendor/github.com/urfave/cli/app.go:276 +0x718
main.main()
        /go/src/github.com/longhorn/longhorn-manager/main.go:57 +0x521
```

**Comment by yasker** [comment]
@cameronbraid Sorry for the late reply somehow missed this one.

Can you try if https://github.com/longhorn/longhorn/issues/957#issuecomment-566195164 works for you?

**Comment by cameronbraid** [comment]
I don't have any PVC's created yet so that workaround doesn't seem to apply

**Comment by cameronbraid** [comment]
I should also note that I dont have any PVs created either

**Comment by yasker** [comment]
Weird, this bug was triggered if the PV doesn't contain `ClaimRef` field, so if there is no PV in the system, this bug shouldn't be triggered. You can take a look at https://github.com/longhorn/longhorn/issues/905 to know more about it.

The code caused the trouble is https://github.com/longhorn/longhorn-manager/blob/96edc510d24a65a6ebb3dbe39480913bf3c61cd4/app/upgrader.go#L33 

And we've fixed it in the upcoming v0.8.0 release: https://github.com/longhorn/longhorn-manager/blob/f5faefaf09fbc25ea3d99cd001c197199be1f488/app/upgrader.go#L37

**Comment by cameronbraid** [comment]
Sorry, I was thinking you meant a longhorn PV.  I do have local volume provisioner PVs

So I expect this fix will fix my issue too.
